cmd/aro/dbtoken.go

```diff
@@ -83,7 +83,13 @@ func dbtoken(ctx context.Context, log *logrus.Entry) error {
 		return err
 	}
 
-	verifier, err := oidc.NewVerifier(ctx, "https://sts.windows.net/"+_env.TenantID()+"/", pkgdbtoken.Resource)
+	// example value: https://login.microsoftonline.com/11111111-1111-1111-1111-111111111111/
+	issuer := _env.Environment().ActiveDirectoryEndpoint + _env.TenantID() + "/"
```

Comment:
I suspect this will not work 
You are chainging to login.microsoftonline.com, which has `v2` at the end and is considered v2 endpoint (application will have to be reconfigured) https://jan-v.nl/post/2020/my-bearer-token-is-sts-windows-net-but-i-need-it-to-be-login-microsoftonline-com/ . This might break dbtoken service, but we dont know this yet as its hard to test
cc @bennerv

Reply:
I was not able to find this application yet to be created so I assume we need to do this manually in prod. 
Hence we need to be aware of required changes.
So if we keel login v2 url (I think we should) we need to update url with `v2` at the end, let @bennerv know and should be fine

Reply:
Reading this https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow im even more confused. I was able to see reference to multiple format url in login.microsoftonline.com domain for the same in different docs :/

Reply:
@mjudeikis I was under the impression that sts.windows.net is now just an alias of login.microsoftonline.com.

- https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/812#issuecomment-456791675
- https://stackoverflow.com/questions/55840510/validating-the-issuer-token-has-issuer-https-login-microsoftonline-com-xv2-0/55840705#55840705

Reply:
@mjudeikis I think this answers the question:

>v1.0 and v2.0 tokens can be issued by both the v1.0 and v2.0 endpoints! id_tokens always match the endpoint they're requested from, and access tokens always match the format expected by the Web API your client will call using that token. So if your app uses the v2.0 endpoint to get a token to call Microsoft Graph, which expects v1.0 format access tokens, your app will receive a token in the v1.0 format.

https://docs.microsoft.com/en-us/azure/active-directory/azuread-dev/azure-ad-endpoint-comparison#token-claims

Reply:
This is broken - MSFT issuer on the JWT is always sts.windows.net and then our OIDC libraries can't validate the JWT token because the expected issuer does not match the actual one.  :(

Reply:
https://docs.microsoft.com/en-us/azure/active-directory/develop/reference-app-manifest#accesstokenacceptedversion-attribute this will fix our problems.  I will document.

Reply:
@bennerv Will this not fix it? https://github.com/Azure/ARO-RP/pull/1611

Reply:
@nilsanderselde  that fixes the dbtoken side, but the dbtoken app registration is still returning a v1 token when the gateway uses it's MSI to request a JWT.  

We can set 2 on the `accessTokenAcceptedVersion` attribute to return a v2 token when requested, but in the v2 token the `aud` claim changes from the redirect URI (https://dbtoken.aro.azure.com/) to the clientID of the app registration.  

https://docs.microsoft.com/en-us/azure/active-directory/develop/id-tokens#payload-claims

Reply:
Talked with MJ and we have a path forward with the change though :+1: